src/java/picard/util/TestNGUtil.java

```diff
@@ -0,0 +1,88 @@
+package picard.util;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.testng.Assert;
+
+import static java.lang.Math.abs;
+
+/**
+ * Small class implementing some utility functions that are useful for test and interfacing with the TestNG framework.
+ */
+public class TestNGUtil {
+    public static abstract class SingleTestUnitTest<TESTCASE_TYPE extends TestNGParameterizable> {
```

Comment:
Convention is to use single letter for formal type parameters or use class name conventions: http://google.github.io/styleguide/javaguide.html#s5.2.8-type-variable-names